src/common/components/chats/chat-context-provider.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useEffect, useState } from "react";
+import { Keys } from "../../../managers/message-manager-types";
+import MessageService from "../../helper/message-service";
+import { useMappedStore } from "../../store/use-mapped-store";
+import { NostrKeysType } from "./types";
+import {
+  createNoStrAccount,
+  getPrivateKey,
+  getUserChatPublicKey,
+  setProfileMetaData
+} from "./utils";
+import * as ls from "../../util/local-storage";
+import { setNostrkeys } from "../../../managers/message-manager";
+
+interface Context {
+  activeUserKeys: NostrKeysType;
+  showSpinner: boolean;
+  revealPrivKey: boolean;
+  chatPrivKey: string;
+  receiverPubKey: string;
+  messageServiceInstance: MessageService | undefined;
+  hasUserJoinedChat: boolean;
+  setRevealPrivKey: (d: boolean) => void;
+  setShowSpinner: (d: boolean) => void;
+  setChatPrivKey: (key: string) => void;
+  setActiveUserKeys: (keys: NostrKeysType) => void;
+  setReceiverPubKey: (key: string) => void;
+  setMessageServiceInstance: (instance: MessageService | undefined) => void;
+  initMessageServiceInstance: (keys: Keys) => MessageService | undefined;
+  joinChat: () => void;
+}
+
+interface Props {
+  children: JSX.Element | JSX.Element[];
+  resetChat: () => void;
+}
+
+export const ChatContext = React.createContext<Context>({
+  activeUserKeys: { pub: " ", priv: "" },
+  showSpinner: false,
+  revealPrivKey: false,
+  chatPrivKey: "",
+  receiverPubKey: "",
+  messageServiceInstance: undefined,
+  hasUserJoinedChat: false,
+  setRevealPrivKey: () => {},
+  setShowSpinner: () => {},
+  setChatPrivKey: () => {},
+  setActiveUserKeys: () => {},
+  setReceiverPubKey: () => {},
+  setMessageServiceInstance: () => {},
+  initMessageServiceInstance: () => (({} as MessageService) || undefined),
+  joinChat: () => {}
+});
+
+export default function ChatContextProvider(props: Props) {
+  const { activeUser } = useMappedStore();
+
+  const [activeUserKeys, setActiveUserKeys] = useState<NostrKeysType>({ pub: " ", priv: "" });
+  const [showSpinner, setShowSpinner] = useState(true);
+  const [chatPrivKey, setChatPrivKey] = useState("");
+  const [revealPrivKey, setRevealPrivKey] = useState(false);
+  const [receiverPubKey, setReceiverPubKey] = useState("");
+  const [messageServiceInstance, setMessageServiceInstance] = useState<MessageService | undefined>(
```

Comment:
As i have set the `messageServiceInstance` to null in `initMessageServiceInstance` function if the `messageServiceInstance` is already in my state.So i think i have to set its default state to "null" instead of "undefined" and I changed and pushed it.